waitress/task.py

```diff
@@ -250,10 +250,12 @@ def close_on_finish():
         # Set the Server and Date field, if not yet specified. This is needed
         # if the server is used as a proxy.
         ident = self.channel.server.adj.ident
-        if not server_header:
-            response_headers.append(('Server', ident))
-        else:
-            response_headers.append(('Via', ident))
+        if ident is not None:
+            if not server_header:
+                response_headers.append(('Server', ident))
+            else:
+                response_headers.append(('Via', ident))
```

Comment:
This should not be removed if ident is not set. We always went to send a `Via` header even if there is no user provided `ident` to make sure that that if waitress is used as a proxy it can't be used to do proxy looping. If there is no user provided ident, we need to make sure to set it to something reasonable like "waitress".

Reply:
I'm okay with dropping the Server header if none is set by the WSGI app, but we should absolutely set a `Via` header if a `Server` header is set by the WSGI app (likely means that waitress is being used as some sort of proxy).

If you also want to drop any potentially proxied `Server` header, I would recommend using some WSGI middleware to drop it from the response before it gets to Waitress, and then Waitress won't add a `Via` header.